### ISSUE TITLE
GT-511C3 fingerprint scanner with raspberry pi 3

### ISSUE THREAD

**Issue by Abhiramianbu** [zafartahirov/fingerpi#9]
Anybody tried it with raspberry pi 3?
How good it is in it ?
I tried using this driver https://github.com/QuickGroup/pyGT511C3 but its is not working.

**Comment by zafartahirov** [comment]
I never tried the sensor with Raspberry Pi 3. Most probably it is the same, assuming the connection you made is correct, and there are no specific settings that enable the Rx/Tx of the serial communication

**Comment by kevex91** [comment]
I have tested it on a Raspberry Pi 3.  

There are a few steps to follow. Check out the full method here: [http://spellfoundry.com/2016/05/29/configuring-gpio-serial-port-raspbian-jessie-including-pi-3/](http://spellfoundry.com/2016/05/29/configuring-gpio-serial-port-raspbian-jessie-including-pi-3/)

It is not as straight forward since ttyS0 has moved in order to make way for the bluetooth module.

Bottom line is that I followed the guide but didn't manage to open a connection to the device using "interactive.py". Demo.py can't get a response when I press "Enter" either. I'll post another comment here if I succeed.

**Comment by zafartahirov** [comment]
According to the link you gave, the port name is different. Did you make sure that you are opening a correct port?

**Comment by kevex91** [comment]
When I run > ls -l /dev
serial0 -> ttyS0
serial1 -> ttyAMA0

I tried changing the port to "serial0" and "ttyS0" in fingerpi.py. In both cases I get this on demo.py:
Opening connection...
Changing baudrate...
Place the finger on the scanner and press <Enter>

Unknown Error occured None

**Comment by kevex91** [comment]
I found the solution for me:

My GPIO pins were in the wrong state for some reason.

I followed this: https://raspberrypi.stackexchange.com/questions/47671/why-my-program-wont-communicate-through-ttyama0-on-raspbian-jessie

To sum it up:
run in terminal > `gpio readall`

check if wPi 15 and 16 are in ALT0 state.

If not then run in terminal and check again > `gpio mode 15 ALT0; gpio mode 16 ALT0`

**Comment by zafartahirov** [comment]
Several things:
1. `demo.py` is just a demo, and is not supposed to be used as a final code.
2. `interactive.py` and `imshow.py` are just a helpers for the `demo.py`
3. The whole process is just creating a `FingerPi` object with the correct parameters, and using its methods to interact with the sensor (it is not well documented -- I might document it someday, otherwise, feel free to send a pull request)

**Comment by ritvikiitkgp** [comment]
Did anyone find the solution? I still can't get mine to work.

**Comment by ritvikiitkgp** [comment]
All right. Got it to work. Thanks guys. And big thanks to @zafartahirov.

**Comment by maheshksingh** [comment]
Hello... Could you please tell me on how did make it work.. I am struggling to make gt-511c3 work with Raspberry pi... Please help

**Comment by maheshksingh** [comment]
It's Raspberry pi3 model b which am using
